.github/workflows.disabled/plagiarism-checker.yml

```diff
@@ -2,90 +2,121 @@
+        PDF_DIR: './pdfs'
+      }
   
   plagiarismCheck:
     name: Check for plagiarism
+    needs: [ convert_md ]
     runs-on: ubuntu-latest
-    needs: [ preflight ]
     steps: 
-    - name: Schedule a check
-      id: schedule_check
+    - name: Get access token
+      id: access_token
+      run: |
+        curl -X POST \
+        https://api.unicheck.com/oauth/access-token \
+          -H 'Content-Type: application/x-www-form-urlencoded' \
+          -d 'grant_type=client_credentials&client_id={{ UNICHECK API KEY }}&client_secret={{ UNICHECK API SECRET }}'
+# Need to store access token provided in response for use in subsequent steps
+    
+    - name: Upload article
+      id: upload_article
       run: |
-        curl https://api.unicheck.com/similarity/checks \
-        -X POST \
-        --output $UPLOAD_RESPONSE_FILENAME \
-        --form data='{"auth":{"api_key": "${{ secrets.UNICHECK_API_KEY }}", "api_secret": "${{ secrets.UNICHECK_API_SECRET }}"}, "wait":true, "lossy":true}' \
-        
+        curl -X POST \
+        "https://api.unicheck.com/files" \
+        -H "Accept: application/vnd.api+json" \
+        -H "Authorization: Bearer {{ access_token }}" \
+        -H "Content-Type: multipart/form-data" -F "file=@{{ path-to-file.pdf }}"
+# Need to store file_id provided in response for use in subsequent step
 
-    - name: Poll status and return results
-      id: status_results
-      run: "https://api.unicheck.com/similarity/checks/{similarity_check_id}"
+    - name: Start similarity check
+      id: start_check
+      run: |
+        curl -X POST \
+        "https://api.unicheck.com/similarity/checks" \
+        -H "Accept: application/vnd.api+json" \
+        -H "Authorization: Bearer {{ access_token }}" \
+        -H "Content-Type: application/vnd.api+json" \
+        -d "{ \"data\": { \"type\": \"similarityCheck\", \"attributes\": { \"search_types\": { \"web\":true, \"library\": false }, \"parameters\": { \"sensitivity\": { \"percentage\": 0, \"words_count\": 8 } } } }, \"relationships\": { \"file\": { \"data\": { \"id\": \"{{ file_id }}\", \"type\": \"file\" } } }}"
+# Need to store similarity_check_id provided in response for use in subsequent steps
       
+    - name: Delay 2 minutes
+      uses: jakejarvis/wait-action@master
+      with:
+        time: '120s'
 
+    - name: Get report
+      id: get_report
+      run: |
+        curl -X POST \
+        https://api.unicheck.com/similarity/checks/{{ similarity_check_id }}/report/export \
+        -H 'Accept: application/vnd.api+json' \
+        -H 'Authorization: Bearer {{ access_token }}' \
+        -H 'Content-Type: application/vnd.api+json' \
+        -d '{
+          "data": {
+          "type" :"similarity-check-report-export",
+          "attributes":{
+            "format" :"pdf",
+            "locale_code" :"EN"
+              }
+            }
+          }'
+# Need to store id provided in response for use in next step
```

Comment:
If you write the output of the curl to a file with `-o`, you could grep for it, and set it as a step variable with `echo -n "::set-output name=similarity_check_report_export_id::"`